src/pages/DeviceDetail/DeviceDetail.js

```diff
@@ -18,27 +19,30 @@ const DeviceDetail = () => {
   const [user, setUser] = useState(null);
   const [error, setError] = useState(null);
 
-  useEffect(async () => {
-    setIsLoading(true);
+  useEffect(() => {
+    const getDevice = async () => {
+      const deviceURI = process.env.REACT_APP_BASE_API_URL + '/devices?id=' + id;
+      const deviceReponse = await fetch(deviceURI);
+      const deviceData = await deviceReponse.json();
+      setDevice(deviceData[0]);
+
+      if (deviceData[0] !== undefined) {
+        await getUser(deviceData[0].userId);
+        setError(null);
```

Comment:
Will this code affect the user experience? The previous error only disappears after the `getUser `is completely done even there is no error.